### PR TITLE
DB-11579 Convert a trigger microbenchmark to a performance IT

### PR DESCRIPTION
**Pull request by OlegMazurov**
New _@Category(Benchmark.class)_.
New _TriggerBenchmark_ is not included in the default IT run (by naming convention).
The benchmark is runnable from IDE against standalone Splice or a remote cluster.

### CODE REVIEW

**Review by hatyo — APPROVED** [review]

**Review by dgomezferro — APPROVED** [review]
It'd be nice to have warmup phases and some other improvements, but this is already great!
@OlegMazurov do you think it'd make sense to integrate it with the JMH project?

File: splice_machine/src/test/java/com/splicemachine/benchmark/TriggerBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                }
+            }
+
+            insert1.close();
+            insert2.close();
+        }
+        catch (Throwable t) {
+            LOG.error("Connection broken", t);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (OlegMazurov) [reply splice_machine/src/test/java/com/splicemachine/benchmark/TriggerBenchmark.java:194]: The differences are, in fact, many: prepared statements, key computation, statistics and error messages. Parameterizing all those would make code less readable. Eliminating many repeating patterns is, however, desirable, so some refactoring is justified. I'm going to revisit this when converting another microbenchmark from DB-10780 to the new framework.

**Comment by OlegMazurov** [comment]
@dgomezferro Will add warmup iterations in next framework update.
Will also reevaluate JMH to see if added value justifies the dependency. My concerns are: flexibility in implementing parallel scenarios with database state validation, variety of collected and reported statistics. Microbenchmarking Java, for which JMH was designed, is qualitatively different from what we consider database "microbenchmarks". However, there is certainly large overlap and some JMH traits are worth borrowing (in whatever form).

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**